### ISSUE TITLE
Rework the Vector Axiom Sink to use Axiom's ingest endpoint instead of the ElasticSearch _bulk endpoint

### ISSUE THREAD

**Issue by darach** [vectordotdev/vector#21292]
### A note for the community

Our vector users are reporting issues with vector axiom sinks that we believe can be avoided
by changing the axiom sink to use the native axiom ingest endpoint instead of the emulated
elasticsearch _bulk endpoint.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Primary reason is to avoid confusion of vector users configuring the axiom sink:
* The behaviour of `_timestamp` differs across elasticsearch and opensearch versions and has
  resulted in changes to the axiom sink to hardcode a solution ( cause was changes to elastics own behaviour itself! )
* Experienced axiom users assume the vector sink to be an axiom ingest based sink.
  + Configuration of query parameters and http headers from this endpoint does not work on
     our elastic _bulk compatible endpoint ( nor should they work on any elastic _bulk compatible endpoint )

### Attempted Solutions

We propose to rework the sink to use the axiom ingest endpoint which will avoid all the above
points of confusion and simplify configuration for our vector users.

### Proposal

We propose that we at Axiom contribute the work and contribute to any future maintenance
as may be needed from time to time. We basically avoid needing ES compatibility for the vector
axiom sink

### References

Useful references

* https://axiom.co/docs/process-data/destinations/elastic-bulk
* https://axiom.co/docs/send-data/ingest
* https://github.com/vectordotdev/vector/issues/12882

One primary source of confusion around `_timestamp` and `@timestamp`. The confusion here is I believe
a result of changes to default behaviour in elasticsearch circa v5.5 on `_timestamp` handling that in turn
cascade to vector's elasticsearch sink, and reached the vector axiom sink in the below PR.

* https://github.com/vectordotdev/vector/pull/16536 

Some documentation impact expected:
* https://axiom.co/docs/send-data/vector#timestamp-field
* The ES compatibility constraint would be lifted/replaced with compatibility on `_time` semantics for axiom.
* However, we would need a migration note for existing users if this proposal is acceptable

TL;DR I think we can avoid all future confusion at least for the vector axiom sink by migrating it to use the ingest endpoint. This in turn reduces maintenance overheads on the elasticsearch endpoint if/when there are changes to elasticsearch as it won't affect the axiom sink once we decouple the sinks so that they can evolve independently

Useful people at Axiom happy to contribute: @darach @Licenser @bahlo 

### Version

v0.41.1

**Comment by jszwedko** [comment]
Thanks for opening this @darach ! So long as folks at Axiom are alright with helping to maintain this sink, I think we'd be fine seeing it rewritten to use the native ingest. We had originally pushed for wrapping the elasticsearch sink mostly to reduce our own maintenance burden.

**Comment by darach** [comment]
Thank you for the quick turnaround @jszwedko - we should have a branch for review soon. I'm on the discord where we could discuss helping with the maintenance and we're happy to do so!
